router/broadcast_router.go

```diff
@@ -15,15 +20,21 @@ type broadcastRouterState struct {
 }
 
 func (state *broadcastRouterState) SetRoutees(routees *actor.PIDSet) {
-	state.routees = routees
+	atomic.StorePointer((*unsafe.Pointer)(unsafe.Pointer(&state.routees)), unsafe.Pointer(&routees))
 }
 
 func (state *broadcastRouterState) GetRoutees() *actor.PIDSet {
-	return state.routees
+	newSet := &actor.PIDSet{}
+	routees := (*actor.PIDSet)(atomic.LoadPointer((*unsafe.Pointer)(unsafe.Pointer(&state.routees))))
+	routees.ForEach(func(i int, pid actor.PID) {
+		newSet.Add(&pid)
+	})
+	return newSet
 }
 
 func (state *broadcastRouterState) RouteMessage(message interface{}) {
-	state.routees.ForEach(func(i int, pid actor.PID) {
+	routees := (*actor.PIDSet)(atomic.LoadPointer((*unsafe.Pointer)(unsafe.Pointer(&state.routees))))
```

Comment:
I think it would be okay to use a non-atomic load on this path, to ensure that routing is not slowed down. The atomic write by the router management actor would ensure that `RouteMessage` always uses a consistent pointer, even if it might temporarily point to an older version of `state.routees`.